src/Mvc/Mvc.ViewFeatures/src/HtmlHelperComponentExtensions.cs

```diff
@@ -37,30 +39,85 @@ public static class HtmlHelperComponentExtensions
         /// <param name="htmlHelper">The <see cref="IHtmlHelper"/>.</param>
         /// <param name="parameters">An <see cref="object"/> containing the parameters to pass
         /// to the component.</param>
+        /// <param name="renderMode">The <see cref="RenderMode"/> for the component.</param>
         /// <returns>The HTML produced by the rendered <typeparamref name="TComponent"/>.</returns>
         public static async Task<IHtmlContent> RenderComponentAsync<TComponent>(
             this IHtmlHelper htmlHelper,
+            RenderMode renderMode,
             object parameters) where TComponent : IComponent
         {
             if (htmlHelper == null)
             {
                 throw new ArgumentNullException(nameof(htmlHelper));
             }
 
-            var httpContext = htmlHelper.ViewContext.HttpContext;
-            var serviceProvider = httpContext.RequestServices;
-            var prerenderer = serviceProvider.GetRequiredService<StaticComponentRenderer>();
+            var context = htmlHelper.ViewContext.HttpContext;
+            return renderMode switch
+            {
+                RenderMode.Server => NonPrerenderedBlazorComponent(context, typeof(TComponent), GetPrameterCollection(parameters)),
+                RenderMode.ServerPrerendered => await PrerenderedBlazorComponentAsync(context, typeof(TComponent), GetPrameterCollection(parameters)),
```

Comment:
nipick - remove blazor from these method names. it doesn't add any additional clarity

Reply:
Replaced blazor with server